### ISSUE TITLE
freepdf-books.com

### ISSUE THREAD

**Issue by uBlock-user** [jspenguin2017/AdBlockProtector#173]
A 10 sec timer runs before download is initiated.

Test-case - http://freepdf-books.com/download/?file=7934 

Solution - `a.timewarp("setInterval", /^1000$/);`

Side-note, I realised `a.timewarp` can be further sped up if you would lower the `ratio` to `0.0002` or even a much smaller value like `0.00000002`, so why not lower the value further making it instanteous ?

**Comment by jspenguin2017** [comment]
It will be faster if you have a better CPU. 
Thanks for the solution, I'll add it to Script.

**Comment by uBlock-user** [comment]
> It will be faster if you have a better CPU.

With the lower ratio value or the current value ?

**Comment by jspenguin2017** [comment]
It depend on if current value exceeds your CPU's capability. 
I know `a.timewarp` isn't efficient, the idea of boosting `setInterval` isn't efficient, but it's just so simple and in many cases it's good enough.

**Comment by uBlock-user** [comment]
At the current value it still takes about a second or two to intialise the download on my end, but if I lower the value to say `0.0000002`, it starts the download before I blink my eye!

**Comment by jspenguin2017** [comment]
I guess you have a better computer... Or the website I initially tested `a.timewarp` was heavier. I initially set the boost ratio to `0.01`, but I realized that it's not much difference than `0.02` (which should be two times difference), so I chose `0.02` to not overload people's CPU. 
There is another reason that I set it to boost by only 50 times by default. In case the boosted code will not stop once the timer is up, 50 times will not burn a hole in system resources.
